desktop_version/src/Graphics.cpp

```diff
@@ -1585,356 +1591,364 @@ void Graphics::drawtrophytext()
 
 void Graphics::drawentities()
 {
-    point tpoint;
-
-    SDL_Rect drawRect;
+    const int yoff = map.towermode ? lerp(map.oldypos, map.ypos) : 0;
 
-#if !defined(NO_CUSTOM_LEVELS)
-    // Special case for gray Warp Zone tileset!
-    int room = game.roomx-100 + (game.roomy-100) * ed.maxwidth;
-    bool custom_gray = INBOUNDS_ARR(room, ed.level)
-    && ed.level[room].tileset == 3 && ed.level[room].tilecol == 6;
-#else
-    bool custom_gray = false;
-#endif
-
-    std::vector<SDL_Surface*> *tilesvec;
-    if (map.custommode && !map.finalmode)
-    {
-        tilesvec = &entcolours;
-    }
-    else
+    if (!map.custommode)
     {
-        tilesvec = &tiles;
-    }
+        for (int i = obj.entities.size() - 1; i >= 0; i--)
+        {
+            if (!obj.entities[i].ishumanoid())
+            {
+                drawentity(i, yoff);
+            }
+        }
 
-    std::vector<SDL_Surface*> *spritesvec;
-    if (flipmode)
-    {
-        spritesvec = &flipsprites;
+        for (int i = obj.entities.size() - 1; i >= 0; i--)
+        {
+            if (obj.entities[i].ishumanoid())
+            {
+                drawentity(i, yoff);
+            }
+        }
     }
     else
     {
-        spritesvec = &sprites;
+        for (int i = obj.entities.size() - 1; i >= 0; i--)
+        {
+            drawentity(i, yoff);
+        }
     }
+}
 
-    int yoff;
-    if (map.towermode)
+void Graphics::drawentity(const int i, const int yoff)
+{
+    if (!INBOUNDS_VEC(i, obj.entities))
     {
-        yoff = lerp(map.oldypos, map.ypos);
+        WHINE_ONCE("drawentity() out-of-bounds!");
+        return;
     }
-    else
+
+    if (obj.entities[i].invis)
     {
-        yoff = 0;
+        return;
     }
 
-    for (int i = obj.entities.size() - 1; i >= 0; i--)
+    point tpoint;
+
+    SDL_Rect drawRect;
+
+#if !defined(NO_CUSTOM_LEVELS)
+    // Special case for gray Warp Zone tileset!
+    const int room = game.roomx-100 + (game.roomy-100) * ed.maxwidth;
+    const bool custom_gray = INBOUNDS_ARR(room, ed.level)
+    && ed.level[room].tileset == 3 && ed.level[room].tilecol == 6;
+#else
+    const bool custom_gray = false;
+#endif
+
+    std::vector<SDL_Surface*>& tilesvec = map.custommode && !map.finalmode ? entcolours : tiles;
```

Comment:
Had a bit of trouble reading this one line:

```suggestion
    std::vector<SDL_Surface*>& tilesvec = (map.custommode && !map.finalmode) ? entcolours : tiles;
```

Reply:
Fixed, rebased, and force-pushed. (Not using GitHub's thing because that adds an unnecessary commit.)